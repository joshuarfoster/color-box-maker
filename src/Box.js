import React from "react";

function Box ({width, height, color, deleteBox, idx}) {

    const handleSubmit = (e) => {
        e.preventDefault();
        deleteBox(idx)
        }

    return (
        <div>
            <div style={{
                width: `${width}px`,
                height: `${height}px`,
                backgroundColor: `${color}`
            }}>
            </div>
            <form onSubmit={handleSubmit}>
                <input type="submit" value='X'/>
            </form>
        </div>
    )
}

export default Box