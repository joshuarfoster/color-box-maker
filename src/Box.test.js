import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", function () {
    render(<Box width='100px' height='100px' color='#ffffff'/>);
});

it("matches snapshot", function () {
    const {asFragment} = render(<Box width='100px' height='100px' color='#ffffff'/>);
    expect(asFragment()).toMatchSnapshot();
})