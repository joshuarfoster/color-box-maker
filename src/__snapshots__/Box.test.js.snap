// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      style="background-color: rgb(255, 255, 255);"
    />
    <form>
      <input
        type="submit"
        value="X"
      />
    </form>
  </div>
</DocumentFragment>
`;
